controllers/drplacementcontrol_controller.go

```diff
@@ -519,11 +588,18 @@ func (r *DRPlacementControlReconciler) processDeletion(ctx context.Context,
 			return ctrl.Result{}, err
 		}
 
-		// Remove DRPCFinalizer. The object will be deleted once
-		// the finalizer is removed
+		// Remove DRPCFinalizer from User PlacementRule.
```

Comment:
This block to remove userPl finalizer should be outside the if block?

I see 2 reasons for this,
1. Order of finalizer addition is userPl followed by DRPC, so removal should be the reverse
2. If userPl finalizer is added, but DRPC finalizer addition errored out and userPl was deleted, we will never remove the finalizer on the userPl

Reply:
good catch.  So the finalizer addition of the uPlRule should be after the finalizer addition of the DRPC.  Then in the removal, we will remove the uPlRule first

Reply:
done

Reply:
So let me think about this as I am writing. 
DRPC is deleted, we remove the usrPlRule finalizer, then let's say we fail to update the DRPC object, so we reconcile again.  If we assume that the usrPlRule is deleted by the time we enter the Reconcile function, then we are stuck in that reconciliation loop failing to get usrPlRule.  True, there is a window of trouble.

Reply:
hmmm, but the reverse is also not good. We remove the DRPC finalizer first, then we fail to remove the usrPlRule.  Doesn't that also leak the usrPlRule?

Reply:
So I pushed a new commit (in order to show the changed differences) to solve the above issue.